### PR TITLE
move windows-amd64 build to a github action

### LINKED ISSUE #1416: windows build: lock to a specific golang version
Currently, we just use whatever the latest version is that's provided by `pacman`. The Linux and Mac builds both pin to a specific golang version. We should do this in Windows as well, for consistency.

This needs to be adjusted in `./windows-bulid.ps1`.

### LINKED ISSUE #1506: ci: Failing Windows builds
**Describe the bug**

Example: https://dev.azure.com/ericxtang/Livepeer%20pipeline/_build/results?buildId=1874&view=logs&j=63c6e092-18c2-5f48-2863-30f858b2a93b&t=042fc50d-cea6-5e09-a550-f624443f682c

Last few lines of failing build:

```
:: Proceed with installation? [Y/n] 
:: Retrieving packages...
downloading msys2-runtime-3.1.4-1-x86_64.pkg.tar.zst...
downloading bash-4.4.023-2-x86_64.pkg.tar.xz...
downloading filesystem-2020.02-2-x86_64.pkg.tar.xz...
downloading mintty-1~3.1.4-1-x86_64.pkg.tar.xz...
downloading pacman-mirrors-20200329-1-any.pkg.tar.xz...
downloading libzstd-1.4.4-2-x86_64.pkg.tar.xz...
downloading zstd-1.4.4-2-x86_64.pkg.tar.xz...
downloading pacman-5.2.1-6-x86_64.pkg.tar.xz...
checking keyring...
checking package integrity...
loading package files...
error: could not open file /var/cache/pacman/pkg/msys2-runtime-3.1.4-1-x86_64.pkg.tar.zst: Child process exited with status 127
error: failed to commit transaction (cannot open package file)
Errors occurred, no packages were upgraded.
Exit code 1
At D:\a\1\s\windows-build.ps1:18 char:5
+     throw "Exit code $LASTEXITCODE"
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Exit code 1:String) [], RuntimeException
    + FullyQualifiedErrorId : Exit code 1
```

### PR DESCRIPTION
**Pull request by iameli**
**What does this pull request do? Explain your changes. (required)**
Our MSYS2 build wasn't working on the Azure Pipeline. This moves us to GitHub Actions, which provide a working MSYS2 environment effortlessly using [https://github.com/marketplace/actions/setup-msys2-environment](numworks/setup-msys2@v1).

Fixes #1506. Also, fixes https://github.com/livepeer/go-livepeer/issues/1416 by pinning us to Go 1.13.

This takes out the `.\windows_build.ps1` PowerShell script because it wasn't working.

I wish GitHub actions had existed when I was setting all this up originally.

**How did you test each of these updates (required)**
* Tested Windows GPU transcoding locally with OBS and this script:

```bash
#!/bin/bash

set -ex

rm -rf livepeer-windows-amd64*
curl "$1" -o livepeer-windows-amd64.tar.gz
unzip livepeer-windows-amd64.tar.gz
cd livepeer-windows-amd64
./livepeer-windows-amd64/livepeer.exe -orchestrator -transcoder -nvidia=0 -serviceAddr=127.0.0.1:44444 -datadir=./orchestrator -cliAddr=0.0.0.0:7935 -testTranscoder=false &
./livepeer-windows-amd64/livepeer.exe -broadcaster -orchAddr 127.0.0.1:44444 -httpAddr 127.0.0.1:8888 -cliAddr 127.0.0.1:7936 &
wait
```

### CODE REVIEW

File: upload_build.sh [hunk 1]
```diff
@@ -51,7 +54,7 @@ if [[ $ARCH == "windows" ]]; then
   FILE=$BASE.zip
   ls /mingw64/bin/
   # This list was produced by `ldd livepeer.exe`
-  LIBS="libffi-7.dll libgcc_s_seh-1.dll libgmp-10.dll libgnutls-30.dll libhogweed-6.dll libiconv-2.dll libidn2-0.dll libintl-8.dll libnettle-8.dll libp11-kit-0.dll libtasn1-6.dll libunistring-2.dll libwinpthread-1.dll zlib1.dll"
+  LIBS="libffi-6.dll libgcc_s_seh-1.dll libgmp-10.dll libgnutls-30.dll libhogweed-4.dll libiconv-2.dll libidn2-0.dll libintl-8.dll libnettle-6.dll libp11-kit-0.dll libtasn1-6.dll libunistring-2.dll libwinpthread-1.dll zlib1.dll"
```
Comment (j0sh) [inline upload_build.sh:57]: Curious where this library downgrade is coming from? Also, I know this was around in the old build, but has it been an issue to have dll dependencies - does this mean the user also needs a particular version of gnutls etc?

Comment (iameli) [reply upload_build.sh:57]: The version of MSYS2 that's installed from the GitHub actions is slightly older than what we were using before, so a few packages are a bit older. The advantage there is that they should fluctuate less, resulting in fewer CI failure package bump chores like https://github.com/livepeer/go-livepeer/pull/1488

It hasn't been an issue 'cause we package all the DLL dependencies in the zip file with the binaries, so there are no external dependencies on the end user as long as they execute livepeer.exe from the folder containing all the DLLs. I'd love to move to a fully-static Windows build someday, but I think that'd mean [building everything on this list from source using PKGBUILD](https://www.msys2.org/wiki/Creating-Packages/) which is not so fun.

**Review by j0sh — APPROVED** [review]
This looks good to me, but I'm not the Windows / CI expert :+1:

**Comment by iameli** [comment]
I don't know that we have one of those 😃

**Comment by j0sh** [comment]
Just a note - the fixup commits here were not actually squashed prior to merging. Wonder if there's a github pre-merge action we can have to shield master from this happening again.

**Comment by yondonfu** [comment]
> Wonder if there's a github pre-merge action we can have to shield master from this happening again.

I think we can use [GH actions](https://help.github.com/en/actions) for this and it looks like there are existing GH actions that can do this. Opened #1520 to track this.

**Comment by iameli** [comment]
Ach, my bad. #1520 looks great.

**Pull request merged**